File: Brute Force/Bubble Sort/code.py
from random import randint

N = 15
MIN = 1
MAX = 20

def BubbleSort(start:int, end:int, array:list):
    newEnd:int = start;
    for i in range(start, end):
        if array[i] > array[i + 1]:
            array[i], array[i+1] = array[i + 1], array[i]
            newEnd = i

    if newEnd == start:
        return
    else:
        BubbleSort(start, newEnd, array)

if __name__ == '__main__':
    array = [randint(MIN, MAX) for _ in range(N)]
    print("Initial Array: ", *array)
    BubbleSort(0, N-1, array)
    print("Sorted Array: ",*array)

File: computer-science/algorithms/Divide and Conquer/Pigeonhole Sort/code.js
const A = Randomize.Array1D({ N: 7 });
const N = A.length;

let min = A[0];
let max = A[0];

for (let i = 1; i < N; i++) {
  if (A[i] < min) {
    min = A[i];
  }
  if (A[i] > max) {
    max = A[i];
  }
}
const range = max - min + 1;

const holes = new Array(range);
for (let i = 0; i < range; i++) {
  holes[i] = [];
}

logTracer.println('Filling up holes');

for (let i = 0; i < N; i++) {

  holes[A[i] - min].push(A[i]);

}

logTracer.println('Building sorted array');

let k = 0;
for (let i = 0; i < range; i++) {
  for (let j = 0; j < holes[i].length; j++) {
    
    A[k++] = holes[i][j];
    
  }
}

logTracer.println(`Sorted array is ${A}`);
